llvm/lib/Target/Hexagon/HexagonTargetMachine.h

```diff
@@ -32,6 +33,9 @@ class HexagonTargetMachine : public LLVMTargetMachine {
                        std::optional<CodeModel::Model> CM, CodeGenOptLevel OL,
                        bool JIT);
   ~HexagonTargetMachine() override;
+  const HexagonSubtarget *getSubtargetImpl() const {
```

Comment:
This method shouldn't exist, subtarget is per-function. You can move `HST->useHVXOps()` check inside the pass'es `runOnMachineFunction` where subtarget is available via MF, and return early if it returns false.

Reply:
Thanks . I will look in to it